pilot/pkg/model/validation.go

```diff
@@ -1813,6 +1815,25 @@ func ValidateVirtualService(_, _ string, msg proto.Message) (errs error) {
 					errs = appendErrors(errs, fmt.Errorf("duplicate hosts in virtual service: %s & %s", hostI, hostJ))
 				}
 			}
+
+			if !strings.Contains(virtualService.Hosts[i], ".") {
+				svcShortNames = append(svcShortNames, virtualService.Hosts[i])
+			} else if strings.Contains(virtualService.Hosts[i], "."+namespace+"."+"svc") {
```

Comment:
are these valid:

a.foo.svc.cluster.local in namespace bar?

a.foo  (no svc.cluster.local suffix)?

Reply:
This is valid.

Reply:
So this check `virtualService.Hosts[i], "."+namespace+"."+"svc"` won't be sufficient then?

I don't understand what is valid and what isn't in VirtualService, and why we need to make this invalid. The current docs we have say nothing about not allowing multiple hosts.

I get the bug that was associated with this, but should we be rejecting these in validation or is there a bug in the config generation we need to fix?

Reply:
Can you look the test cases i added, it will help.